File: core-service/src/main/java/com/sandjelkovic/dispatchd/domain/data/repository/SeasonRepository.java
package com.sandjelkovic.dispatchd.domain.data.repository;


import com.sandjelkovic.dispatchd.domain.data.entity.Season;
import com.sandjelkovic.dispatchd.domain.data.entity.TvShow;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface SeasonRepository extends CrudRepository<Season, Long> {

	List<Season> findByTvShow(TvShow show);
}

File: content-service/src/main/kotlin/com/sandjelkovic/dispatchd/contentservice/data/repository/SeasonRepository.kt
package com.sandjelkovic.dispatchd.contentservice.data.repository

import com.sandjelkovic.dispatchd.contentservice.data.entity.Season
import com.sandjelkovic.dispatchd.contentservice.data.entity.Show
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository

/**
 * @author sandjelkovic
 * @date 27.1.18.
 */
interface SeasonRepository : PagingAndSortingRepository<Season, Long> {
    fun findByShow(show: Show): List<Season>

    fun findByShow(show: Show, pageable: Pageable): Page<Season>
}
